applications/FluidDynamicsApplication/tests/cpp_tests/test_two_fluid_navier_stokes_element.cpp

```diff
@@ -2086,5 +2086,122 @@ namespace Kratos {
             KRATOS_CHECK_VECTOR_NEAR(reference_RHS, RHS, 1e-2);
         }
 
+        KRATOS_TEST_CASE_IN_SUITE(ElementTwoFluidNavierStokesCut3D4NMomentomCorrection, FluidDynamicsApplicationFastSuite)
+        {
+            Model current_model;
+            ModelPart& modelPart = current_model.CreateModelPart("Main");
+            modelPart.SetBufferSize(3);
+
+            // Variables addition
+            modelPart.AddNodalSolutionStepVariable(BODY_FORCE);
+            modelPart.AddNodalSolutionStepVariable(DENSITY);
+            modelPart.AddNodalSolutionStepVariable(DYNAMIC_VISCOSITY);
+            modelPart.AddNodalSolutionStepVariable(DYNAMIC_TAU);
+            modelPart.AddNodalSolutionStepVariable(PRESSURE);
+            modelPart.AddNodalSolutionStepVariable(VELOCITY);
+            modelPart.AddNodalSolutionStepVariable(MESH_VELOCITY);
+            modelPart.AddNodalSolutionStepVariable(DISTANCE);
+
+            // Process info creation
+            double delta_time = 0.1;
+            modelPart.GetProcessInfo().SetValue(DYNAMIC_TAU, 0.001);
+            modelPart.GetProcessInfo().SetValue(DELTA_TIME, delta_time);
+            Vector bdf_coefs(3);
+            bdf_coefs[0] = 3.0/(2.0*delta_time);
+            bdf_coefs[1] = -2.0/delta_time;
+            bdf_coefs[2] = 0.5*delta_time;
+            modelPart.GetProcessInfo().SetValue(BDF_COEFFICIENTS, bdf_coefs);
+            modelPart.GetProcessInfo().SetValue(VOLUME_ERROR, 0.0);
+            modelPart.GetProcessInfo().SetValue(MOMENTUM_CORRECTION, true);
+
+            // Set the element properties
+            Properties::Pointer pElemProp = modelPart.CreateNewProperties(0);
+            pElemProp->SetValue(DENSITY, 1000.0);
+            pElemProp->SetValue(DYNAMIC_VISCOSITY, 1.0e-05);
+            NewtonianTwoFluid3DLaw::Pointer pConsLaw(new NewtonianTwoFluid3DLaw());

```

Comment:
```suggestion
            auto p_cons_law = Kratos::make_shared<NewtonianTwoFluid3DLaw>();
```
Please don't use the old syntax.

As a comment aside, these tests need a major clean up (everything is copied and pasted from one test to the next).